### PR TITLE
Release pyDARNio v1.2.0

### PR DESCRIPTION
**Pull request by carleyjmartin**
# Scope 

pyDARNio is having a new release! 
This is a minor release which includes:
- Handling of extra fields in FitACF 3.0 coming in the next RST release
- More flexibility around optional fields for all file types
- Updates and memory improvements for Borealis file io

## Approval

**Number of approvals:** 3 or more including code review and testing
**DEADLINE FOR TESTING: ~~14th NOV~~ 25th NOV**

## Test
- Each common file type (rawacf, fitacf, grid, map) should be tested for reading and writing if possible (read into a dictionary and write straight back to file).
- I will append this PR later with some data given to test the Borealis conversion and reading - More information from @RemingtonRohel if required, but this testing should be done by a couple of the Borealis engineers if possible. 
- Test integration into pyDARN for anyone in the DVWG.
- Check the zenodo json list and let me know if I'm missing anyone and if your ORCiD is correct.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,14 +9,12 @@ Python data IO library for the Super Dual Auroral Radar Network (SuperDARN).
...
-pyDARNio is released! This is a patch release to address the following issues:
-- Bug fix with initializing empty arrays when converting HDF5 files from site- to array-structured
-- Bug fixes with converting files from HDF5 to DMAP
-  - correctly check blanked_samples
-  - support multiple beams per record
-  - replacement of far-range lag0 data in rawacf conversion
+## Version 1.2.0 - Release!
+
+pyDARNio has a new release! This is a minor release which includes:
+- Handling of extra fields in FitACF 3.0 coming in the next RST release
```
Comment (ecbland) [inline README.md:15]: Two of the extra fields will be included in all `fitacf`-format files, not just those produced by FitACF3.0, so I've added some more detail to the description to clarify:

```suggestion
- Handling of extra fields in fitacf files coming the next RST release, including the name of the ACF fitting algorithm used (`algorithm`), the value of TDIFF used to calculate the elevation angles (`tdiff`), and more descriptive field names for the elevation angle fields in FitACF 3.0 (`elv_error` & `elv_fitted`)
```

**Comment by RemingtonRohel** [comment]
Theo and I have uncovered a couple of bugs with restructuring Borealis files (#50 and #51), so I am planning on trying to fix those bugs today then test this PR finally.

**Comment by carleyjmartin** [comment]
[test_files.zip](https://github.com/SuperDARN/pyDARNio/files/10068735/test_files.zip)
Included here are a map file, a grid file, an old fitacf3 file, an old fitacf2.5 file and two new test files with the new formatting for testing. All should read in with no issues. Thanks! :)

Here's some code for rough testing for reading and writing the fitacf files, the same method applies for maps and grids but with read_{filetype} instead. I usually import pyDARN and plot it with that to test but that's not needed here as long as there's no errors.
```python
import pydarnio

fitacf2_file = "/Users/carley/Documents/data/fitacf25/20220124.1200.00.kod.d.fitacf"
fitacf3_file = "/Users/carley/Documents/data/fitacf3/20220108.0400.00.sas.a.fitacf"
#newfitacf3_file = "/Users/carley/Desktop/modified_format.lyr.fitacf3"
newfitacf3_file="/Users/carley/Desktop/test_file.han.fitacf3"

fitacf2_read = pydarnio.SDarnRead(fitacf2_file)
fitacf3_read = pydarnio.SDarnRead(fitacf3_file)
newfitacf3_read = pydarnio.SDarnRead(newfitacf3_file)

fitacf2_data = fitacf2_read.read_fitacf()
fitacf3_data = fitacf3_read.read_fitacf()
newfitacf3_data = newfitacf3_read.read_fitacf()

print('Fitacf 2')
print(fitacf2_data[0])
print('Fitacf 3')
print(fitacf3_data[0])
print('New format Fitacf 3')
print(newfitacf3_data[0])

write1 = "/Users/carley/Desktop/write_test.kod.fitacf2"
write2 = "/Users/carley/Desktop/write_test.sas.fitacf3"
write3 = "/Users/carley/Desktop/write_test.han.fitacf3"
# Write test
fitacf2_write = pydarnio.SDarnWrite(fitacf2_data, write1).write_fitacf(write1)
fitacf3_write = pydarnio.SDarnWrite(fitacf3_data, write2).write_fitacf(write2)
newfitacf3_write = pydarnio.SDarnWrite(newfitacf3_data, write3).write_fitacf(write3)
```

File: .zenodo.json [hunk 2]
```diff
@@ -1,12 +1,22 @@
 {
     "creators":[
         {
-            "name": "SuperDARN Data Standards Working Group"
+            "name": "SuperDARN Data Visualization Working Group"
         },
         {
             "affiliation": "University of Saskatchewan",
             "name": "Rohel, R.A."
```
Comment (RemingtonRohel) [inline .zenodo.json:8]:
```suggestion
            "name": "Rohel, R.A.",
            "orcid": "0000-0003-2208-1553"
```

**Review by RemingtonRohel — APPROVED** [review]
Tested with Carley's files, was able to read/write all fitacf2.5, fitacf3, grid, and map files.

Also tested reading/writing/restructuring Borealis files, everything seems to be working smoothly.

**Comment by Shirling-VT** [comment]
Python 3.7.13
MacOS Monterey 12.5.1

Tested with Carley's files and some of my own fitacf bz2 files, was able to read/write all fitacf2.5, fitacf3, grid, and map files with the following codes slightly modified from Carley's codes. They are all working fine. I tested it directly using pydarnio and haven't tested in pydarn by importing pydarnio yet. I didn't tested reading/writing/restructuring Borealis files since I don't have any of them.

import pydarnio
import bz2

###########################Read test#################################
fitacf2_file = "/Users/xueling/data/superdarn/2016/fitacf/fhe/20160125.0002.00.fhe.fitacf.bz2"
with bz2.open(fitacf2_file) as fp:
    fitacf2_stream = fp.read()
fitacf2_read = pydarnio.SDarnRead(fitacf2_stream,True)
fitacf2_data = fitacf2_read.read_fitacf()
print('Fitacf 2')
print(fitacf2_data[0])


fitacf3_file = "/Users/xueling/data/superdarn/2017/fitacf3/bks/20170528.0002.00.bks.fitacf3.bz2"
with bz2.open(fitacf3_file) as fp:
    fitacf3_stream = fp.read()
fitacf3_read = pydarnio.SDarnRead(fitacf3_stream,True)
fitacf3_data = fitacf3_read.read_fitacf()
print('Fitacf 3')
print(fitacf3_data[0])

newfitacf3_file="/Users/xueling/pydarnio_test/pr_49/test_files/test_file.han.fitacf3"
newfitacf3_read = pydarnio.SDarnRead(newfitacf3_file)
newfitacf3_data = newfitacf3_read.read_fitacf()
print('New format Fitacf 3')
print(newfitacf3_data[0])

newfitacf3_file1="/Users/xueling/pydarnio_test/pr_49/test_files/modified_format.lyr.fitacf3"
newfitacf3_read1 = pydarnio.SDarnRead(newfitacf3_file1)
newfitacf3_data1 = newfitacf3_read1.read_fitacf()
print('New format Fitacf 3')
print(newfitacf3_data1[0])


grid_file = "/Users/xueling/pydarnio_test/pr_49/test_files/20011220.8000.grid"
grid_read = pydarnio.SDarnRead(grid_file)
grid_data = grid_read.read_grid()
print('Grid data')
print(grid_data[0])

map_file = "/Users/xueling/pydarnio_test/pr_49/test_files/20190106.n.map"
map_read = pydarnio.SDarnRead(map_file)
map_data = map_read.read_map()
print('Map data')
print(map_data[0])


###########################Write test#################################
write1 = "/Users/xueling/Desktop/write_test.fhe.fitacf2"
write2 = "/Users/xueling/Desktop/write_test.bks.fitacf3"
write3 = "/Users/xueling/Desktop/write_test.han.fitacf3"
write4 = "/Users/xueling/Desktop/write_test.lyr.fitacf3"
write5 = "/Users/xueling/Desktop/write_test.grid"
write6 = "/Users/xueling/Desktop/write_test.map"

fitacf2_write = pydarnio.SDarnWrite(fitacf2_data, write1).write_fitacf(write1)
fitacf3_write = pydarnio.SDarnWrite(fitacf3_data, write2).write_fitacf(write2)
newfitacf3_write = pydarnio.SDarnWrite(newfitacf3_data, write3).write_fitacf(write3)
newfitacf3_write1 = pydarnio.SDarnWrite(newfitacf3_data1, write4).write_fitacf(write4)
grid_write = pydarnio.SDarnWrite(grid_data, write5).write_grid(write5)
map_write = pydarnio.SDarnWrite(map_data, write6).write_map(write6)

**Comment by carleyjmartin** [comment]
With two reviews in now, I'm intending to release this branch early next week, thanks guys!

**Pull request merged**